src/classes/CON_Email.cls

```diff
@@ -281,11 +281,24 @@ public class CON_Email {
                 'Work Email' => 'Work',
                 'Alternate Email' => 'Alternate'
             };
-
+                     
+            String namespace = UTIL_Namespace.getNamespace();
+            String prefix = (String.isNotBlank(namespace)) ? namespace + '__' : '';
+            List<Schema.PicklistEntry> preferredEmailPicklist = UTIL_Describe.getFieldPicklistEntries('Contact', prefix+'Preferred_Email__c');
+            List<String> preferredEmailValues = new List<String>();
+            
+            for (Schema.PicklistEntry eachValue: preferredEmailPicklist) {
+                preferredEmailValues.add(eachValue.getValue());
+            }
+             
             this.value = val;
             this.label = lbl;
             this.apiName = api;
-            this.prefLabel = (preferredEmailMappings.containsKey(lbl)) ? preferredEmailMappings.get(lbl) : lbl;
+            if (!preferredEmailValues.contains(lbl)) {
```

Comment:
@skosgi1 Some comments explaining this might be ideal. Something like: "Use the old preferred email mappings if the picklist values do not contain the field label"?

Or, maybe update the comment on line 278 "Store default email field mappings to preferred email picklist values prior to EDA 1.86."

Reply:
Good point. Updated it.